chromium_src/url/url_util.cc

```diff
@@ -0,0 +1,43 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "url/third_party/mozilla/url_parse.h"
+#include "url/url_canon.h"
+
+namespace url {
+struct Component;
+}
+
+namespace {
+const char kIPFSScheme[] = "ipfs";

```

Comment:
AFAIU we are introducing a discrepancy, because such urls are not "standard", but nevertheless have hosts, correct?

Reply:
only in case of CIDv0 we have to skip canonicalization because they must have Capital letters in a hash( Qm..), in this case we just copy it as is for host part, others hashes meet url standards and can be used as usual hosts

Reply:
I mean, `IsStandard()` that depends on the schema

Reply:
I see, added it as Standard to make GetOrigin worked as well

Reply:
@bridiver I added couple of new patches here, I think you need to check it too